src/core/server/saved_objects/migrationsv2/integration_tests/migration_from_v1.test.ts

```diff
@@ -118,18 +118,23 @@ describe('migration v2', () => {
       .getTypeRegistry()
       .getAllTypes()
       .reduce((versionMap, type) => {
-        if (type.migrations) {
-          const migrationsMap =
-            typeof type.migrations === 'function' ? type.migrations() : type.migrations;
-          const highestVersion = Object.keys(migrationsMap).sort(Semver.compare).reverse()[0];
+        const { name, migrations, convertToMultiNamespaceTypeVersion } = type;
+        if (migrations || convertToMultiNamespaceTypeVersion) {
+          const migrationsMap = typeof migrations === 'function' ? migrations() : migrations;
+          const migrationsKeys = migrationsMap ? Object.keys(migrationsMap) : [];
+          if (convertToMultiNamespaceTypeVersion) {
+            // Setting this option registers a conversion migration that is reflected in the object's `migrationVersions` field
+            migrationsKeys.push(convertToMultiNamespaceTypeVersion);
+          }
```

Comment:
QQ: so you're cheating for `convertToMultiNamespaceTypeVersion` 2 times. if think the first is redundant?

Reply:
`convertToMultiNamespaceTypeVersion` will impact the object's `migrationVersion` regardless of whether a `migrations` object/function is defined or not.

To rephrase: the union of the object type definition's `migrations` and `convertToMultiNamespaceTypeVersion` determines the resulting objects' `migrationVersion`.

Reply:
@kertal in 122 line it's `||` operation so all good :)